tests/Data/ProgrammesDb/EntityRepository/PromotionRepository/FindPromotionsByPidTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Tests\BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\CoreEntityRepository;
+
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use Tests\BBC\ProgrammesPagesService\AbstractDatabaseTest;
+
+/**
+ * @coversDefaultClass BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\PromotionRepository
+ */
+class FindPromotionsByPidTest extends AbstractDatabaseTest
+{
+    private $promotionRepository;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->loadFixtures(['PromotionsFixture']);
+        $this->promotionRepository = $this->getRepository('ProgrammesPagesService:Promotion');
+    }
+
+    /**
+     * @covers ::findActivePromotionsByPid
+     */
+    public function testOnlyActivePromotionsAreReceived()
+    {
+        $brandPid = new Pid('b010t19z');
+        $dbPromotions = $this->promotionRepository->findActivePromotionsByPid($brandPid, 300, 0);
+
+        $this->assertEquals(['p000000h', 'p000001h', 'p000004h'], array_column($dbPromotions, 'pid'));
+        $this->assertCount(1, $this->getDbQueries());
+    }
+
+    /**
+     * @covers ::findActiveSuperPromotionsByPid
+     */
+    public function testOnlyActiveSuperpromotionsAreReceivedACoreEntity()
```

Comment:
Convention is to have one TestCase per function in the repository, named after the function you're testing. Move things that test `findActiveSuperPromotionsByPid` into a class called `FindActiveSuperPromotionsByPidTest.php`